graphbolt/src/macro.h

```diff
@@ -0,0 +1,51 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file graphbolt/src/macro.h
+ * @brief Common macros for graphbolt package.
+ */
+
+#ifndef GRAPHBOLT_MACRO_H_
+#define GRAPHBOLT_MACRO_H_
+
+#include <torch/torch.h>
+
+/**
+ * Dispatch according to torch scalar type.
+ *
+ * ATEN_ID_TYPE_SWITCH(type_per_edge.dtype(), DataType, {
+ *   // Now DataType is the type corresponding to data type in type_per_edge.
+ *   // For instance, one can do this for a CPU array:
+ *   DataType *data = static_cast<DataType *>(type_per_edge.data_ptr());
+ * });
+ */
+#define ATEN_DATA_TYPE_SWITCH(val, DataType, ...)                           \
```

Comment:
just curious, please feel free to ignore this comment. personally I don't like this macro which define a `DataType` in runtime with multiple `if-else`. I am wondering if we could avoid this macro and make calls that rely on `DataType` to be template-like which works well with `tensor.dtype()` automatically.